Search/Factory.php

```diff
@@ -49,13 +53,37 @@ public function makeField($name, $value, $type = Field::TYPE_STRING)
     }
 
     /**
-     * Make a new metadata object representing a mapped
-     * searchable class
+     * Make a new metadata object repesenting a mapped
+     * searchanle class
+     */
+    public function makeClassMetadata($class)
+    {
+        return new ClassMetadata($class);
+    }
+
+    /**
+     * Make a new metadata object representing an index for
+     * a mapped searchable class.
      *
      * @return IndexMetadata
      */
-    public function makeIndexMetadata($class)
+    public function makeIndexMetadata()
+    {
+        return new IndexMetadata();
+    }
+
+    public function makeMetadataField($name)
+    {
+        return new MetadataField($name);
+    }
+
+    public function makeMetadataProperty($path)
```

Comment:
I just checked a few factory classes in Symfony, and they are using create. I would prefer to use this word, since it is more common, and what people expect.